robottelo/api/utils.py

```diff
@@ -437,3 +438,34 @@ def configure_provisioning(org=None, loc=None):
         'host_group': host_group.name,
         'domain': domain.name,
     }
+
+
+def get_role_by_bz(bz_id):
+    """Create and configure custom role entity for the testing of specific bugs
+     This function will read the dictionary of permissions and their associated
+     bugzilla id's from robottelo.constants "PERMISSIONS_WITH_BZ",
+     these permissions will create filter and a single role will be created
+     from all the filters.
+
+     :param bz_id: This is the bugzilla id that is specified in the
+        PERMISSIONS_WITH_BZ list, all the permissions associated with the bz_id
+        will be fetched and filters will be created
+     :return: A single role entity will be created from all the created filters
+     """
+    perms_dict = {}
+    role = entities.Role().create()
+    for res_type in PERMISSIONS_WITH_BZ.keys():
+        for value in PERMISSIONS_WITH_BZ[res_type]:
```

Comment:
Just a nitpick that you @san7ket can decide to change or not:

For this kind of Cartesian product I prefer using the function to avoid nested loops whyle typing less boilerplate:

```python
>>> from itertools import product
>>> product([1,2], 'AB')
<itertools.product object at 0x7f7ad96bbeb0>
>>> list(_)
[(1, 'A'), (1, 'B'), (2, 'A'), (2, 'B')]
>>> for n, c in product([1,2], 'AB'): print(n, c)
... 
(1, 'A')
(1, 'B')
(2, 'A')
(2, 'B')
```
This function accepts accepts any iterable ;)